### PR TITLE
Removed toggling dev mode

### LINKED ISSUE #877: Tearing down fabric runtime in dev mode does not reset the dev mode icon
## Description
Tearing down fabric runtime does not reset the dev mode status of the peer/gateway, so when the fabric runtime is restarted, the UI reports being in the wrong mode 

## Expected Behavior

There are 2 options:
1. The fabric runtime is always started in 'normal' mode and the UI reflects that
2. The fabric runtime starts in the same mode as the last runtime to be torn down

## Your Environment

* IBM Blockchain Platform Extension Version:
* Visual Studio Code version: 1.33.1
* Environment name and version Node.js v8.15.0, npm v6.4.1
* Operating System and version: MacOS 10.14.4

### PR DESCRIPTION
**Pull request by Jakeeyturner**
Close #877 and #928 
Signed-off-by: Jake Turner <jaketurner25@live.com>

### CODE REVIEW

**Comment by heatherlp** [comment]
I can't remember where this is at, but if we're keeping dev mode - can we update the peer tooltip to explain it's in dev mode when the infinity symbol is there? Helps explain what the infinity symbol actually is :)

File: client/src/fabric/FabricConnection.ts [hunk 1]
```diff
@@ -197,4 +203,8 @@ export abstract class FabricConnection {
     private getAllPeers(): Array<Client.Peer> {
         return this.gateway.getClient().getPeersForOrg();
     }
+
+    private getClientTimeout(): number {
```
Comment (cazfletch) [inline client/src/fabric/FabricConnection.ts]: This shouldn't be done in here, it should be in the connect command and you pass the timeout through

File: client/src/debug/FabricDebugConfigurationProvider.ts [hunk 2]
```diff
@@ -50,26 +51,26 @@ export abstract class FabricDebugConfigurationProvider implements vscode.DebugCo
                 return;
             }
 
-            if (!this.runtime.isDevelopmentMode()) {
+            const developmentMode: boolean = await this.runtime.isDevelopmentMode();
```
Comment (cazfletch) [inline client/src/debug/FabricDebugConfigurationProvider.ts]: i think we can take this out. We need to update the generator version anyway so if we update the version here then they will have to tear down anyway and then always be in dev mode

File: client/src/fabric/FabricRuntime.ts [hunk 3]
```diff
@@ -321,13 +321,21 @@ export class FabricRuntime extends EventEmitter {
         await this.killChaincodeInner(args, outputAdapter);
     }
 
-    public isDevelopmentMode(): boolean {
-        return this.developmentMode;
-    }
+    public async isDevelopmentMode(): Promise<boolean> {
```
Comment (cazfletch) [inline client/src/fabric/FabricRuntime.ts]: see above comment about not needing this

File: client/src/util/MetadataUtil.ts [hunk 4]
```diff
@@ -75,6 +75,7 @@ export class MetadataUtil {
 
             if (checkForEmpty && (contractsMap.size === 0)) {
                 outputAdapter.log(LogType.ERROR, `No metadata returned. Please ensure this smart contract is developed using the programming model delivered in Hyperledger Fabric v1.4+ for JavaScript and TypeScript`);
+                return null;
```
Comment (cazfletch) [inline client/src/util/MetadataUtil.ts]: you don't need the null bit

File: client/README.md [hunk 5]
```diff
@@ -122,6 +122,19 @@ The key `ecert` must be set to true in order for a smart contract to be able to
...
 Hovering over an identity in the `Fabric Wallets` panel will show any attributes associated with the identity.
 
+### Changing transaction timeout values
+
+In the extension, the default timeout value for transactions is 120 seconds.
+However, when debugging a transaction it might be necessary to change the time taken before a transaction fails.
+
+The following settings can be changed in the user settings to increase or decrease the time taken for a transaction to timeout:
+- `ibm-blockchain-platform.fabric.client.timeout` - timeout value used for communication to the peers and orderers
+- `ibm-blockchain-platform.fabric.chaincode.timeout` - timeout value for chaincode that's been started by the Local Fabric
```
[the remark this answers is not in the document]
Comment (Jakeeyturner) [reply client/README.md]: Updated the README to mention the two different variables we change

File: client/src/fabric/FabricConnection.ts (hunk 1 above)
Comment (Jakeeyturner) [reply client/src/fabric/FabricConnection.ts]: Removed this function and now pass in the timeout value in the connect command (when calling connect)

File: client/src/debug/FabricDebugConfigurationProvider.ts (hunk 2 above)
Comment (Jakeeyturner) [reply client/src/debug/FabricDebugConfigurationProvider.ts]: Removed all signs of development mode!

**Pull request merged**